### PR TITLE
test: Port more query tests to kafkaless

### PR DESCRIPTION
**Pull request by carols10cents**
Part of https://github.com/influxdata/idpe/issues/17076

The first commit just ports the tests; the second commit un-nests the tests from the module and is probably easier to review without whitespace changes.

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_cases/querier.rs [hunk 1]
```diff
@@ -1,898 +1,220 @@
...
-        test_helpers::maybe_start_logging();
-        let database_url = maybe_skip_integration!();
-
-        let table_name = "the_table";
-
-        // Set up the cluster  ====================================
-        let router_config = TestConfig::new_router(&database_url);
-        // can't use standard mini cluster here as we setup the querier to panic
-        let ingester_config = TestConfig::new_ingester(&router_config)
-            .with_ingester_flight_do_get_panic(2)
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier.rs:83]: This CLI flag that tells the ingester to panic a certain number of times is... not a great way to test, IMO. I'd rather not recreate this in ingester2. 

I'm not actually sure what these tests are concerned with, actually-- the behavior of the querier if the ingester panics? the `query-ingester` command (in `repeat_ingester_request_based_on_querier_logs`)? I decided these weren't worth porting.

Comment (alamb) [reply influxdb_iox/tests/end_to_end_cases/querier.rs:83]: > I'm not actually sure what these tests are concerned with, actually-- the behavior of the querier if the ingester panics? the query-ingester command (in repeat_ingester_request_based_on_querier_logs)? I decided these weren't worth porting.

Yes, the point was to ensure that the querier didn't get wedged when an ingester panic'd. Given that we have currently gone with "ingester2 process will abort immediately on a panic" (see https://github.com/influxdata/influxdb_iox/pull/6788 from @domodwyer ) I agree there is nothing here to port over

File: influxdb_iox/tests/end_to_end_cases/querier.rs [hunk 2]
```diff
@@ -375,14 +246,262 @@ mod with_kafka {
...
+        let mut cluster = MiniCluster::create_shared2_never_persist(database_url).await;
+
+        let steps = vec![
+            Step::RecordNumParquetFiles,
+            // create UNPERSISTED ingester data
+            //
+            // IMPORTANT: The data MUST NOT be persisted before the first query is executed, because persistence
+            //            calculates the partition sort key. The original bug was that the first query on a completely
+            //            unpersisted partition would cache the NULL/None sort key which would later lead to a panic.
+            Step::WriteLineProtocol(format!("{table_name},tag=A val=\"foo\" 1")),
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_cases/querier.rs:445]: I think the original test had some way here to verify the data was not persisted. 

```
            Step::AssertNotPersisted,
```

Could we do something similar like this perhaps?

```
            // Ensure there is no peristed files?
            Step::WaitForPersisted2 {
                expected_increase: 0,
            },
```

**Review by alamb — APPROVED** [review]
I had one small comment, which I think should be get a response prior to merge but otherwise looks good to me 👌  Thanks @carols10cents

File: influxdb_iox/tests/end_to_end_cases/querier.rs (hunk 2 above)
Comment (carols10cents) [reply influxdb_iox/tests/end_to_end_cases/querier.rs:445]: 😬 I don't think that'll work. Say we check for the number of parquet files, and there indeed is an increase of 0 so we continue. Then, concurrently and incorrectly, the ingester persists the data, and the test proceeds to the query step.

I think this test has to trust that `MiniCluster::create_shared2_never_persist` does what it says it does, as long as you don't use `Step::Persist`, and [we do have a test for that](https://github.com/influxdata/influxdb_iox/blob/8ab71d08ba87e07391dce36a2a7e6c257279f5a0/influxdb_iox/tests/end_to_end_cases/querier.rs#L693-L714).

**Pull request merged**